addons/crm/models/crm_lead.py

```diff
@@ -1238,14 +1273,13 @@ def _pls_get_naive_bayes_probabilities(self, batch_mode=False):
         leads_values = set()
         won_leads = set()
         for lead_id, values in leads_values_dict.items():
-            for couple in values['values']:
-                if couple[0] == 'stage_id' and couple[1] in won_stage_ids:
+            for field, value in values['values']:
+                if field == 'stage_id' and value in won_stage_ids:
                     won_leads.add(lead_id)
-                leads_values.add(couple)
-
+                leads_values.add((field, value))
         # get all variable related records from frequency table, no matter the team_id
-        fields = list(set([lead_value[0] for lead_value in leads_values]))
-        frequencies = LeadScoringFrequency.search([('variable', 'in', fields)], order="team_id asc")
+        leads_fields = list(set([lead_value[0] for lead_value in leads_values]))
```

Comment:
You build leads_values as a container of tuples, but it seems like you only use the first value of every tuple?
Why not making i a regular list/set then?